internal/permission/implementation.go

```diff
@@ -1,12 +1,12 @@
 package permission
 
 type DataProvider interface {
-	get(fqfields []Fqfield) map[Fqfield]Value
+	Get(fqfields []Fqfield) map[Fqfield]Value
 }
 
-type Fqfield = string
-type Value = interface{}
+type Fqfield string
+type Value interface{}
```

Comment:
I would suggest, that you do not define this types but use `string` and `interface{}` directly. This makes it much easier for the callers of you interface. Also, it is not possible to convert a variable of type `map[string]interface{}` to `map[permission.Fqfield]permission.Value` directly. This is a O(n) operation:

```
d1 := make(map[string]interface{})
// set some values
d2 := make(map[permission.Fqfield]permission.Value, len(d1))
for k, v := range d1 {
  d2[k] = v
}
```